### PR TITLE
fixed the building error caused by the Temp directory was not set in OSX

### PR DESCRIPTION
**Pull request by kerryjiang**
The related issue:
https://github.com/OrchardCMS/Orchard2/issues/702

### CODE REVIEW

**Comment by Jetski5822** [comment]
@jtkech Can you review?

**Comment by sebastienros** [comment]
I would personally use a `.build` folder. Knowing that another subfolder will be created. @kerryjiang wait for more feedback before changing it.

**Comment by jtkech** [comment]
@Jetski5822 ok, i will take a look on what @kerryjiang did and which looks good.

@sebastienros i will think about this.

Maybe the best would be to get rid of the Temp folder. Working on watching modules razor files, i saw that, when building a main project, project references can be resolved (here would be good to filter modules) and we can call on them a msbuild task by passing custom properties (e.g ApplicationDirectory). So module projects could copy their asset files directly in the main app Packages folder.

Let me some time to see if it is doable.

**Comment by jtkech** [comment]
So, i have a solution without using the `Temp` folder.

- As said, after building the main app can call on each project reference our `CopyPackageAssetFiles` task. One problem was that msbuild is showing errors on projects which don't have this task. And there is no simple way to check if a project has a given target task.

- I could use the `AfterBuild` task that each project has by default, and where modules would copy their assets. But i don't like it, a project could have a custom `AfterBuild` only intended to be used after building it (not the main app).

- So the strategy is that after building a module, the module only checks if the manifest file exists, if not it creates it but put it in its project directory, not in the `Temp` folder.

- Then, after building the main app, the main app call `CopyPackageAssetFiles` on each project references, but only those where a manifest file (`Module.txt / Theme.txt`) exists. Then, because we pass the `ApplicationDirectory` as a property, each module can copy its assets directly in the main app.

I will do a PR to show this solution, **but, to avoid conflicts, maybe better to wait for the watching PR #686 which modify the same `.props` and `.targets` files**.

File: src/OrchardCore/OrchardCore.AsModule/OrchardCore.AsModule.props [hunk 1]
```diff
@@ -8,6 +8,7 @@
   <PropertyGroup>
     <ModuleManifest>Name: $(MSBuildProjectName)</ModuleManifest>
     <ExcludedFiles>**\*.cs;*.csproj*;obj\**;bin\**;Assets.json;Assets\**;Properties\**;*.props;*.targets;.vs\**</ExcludedFiles>
+    <Temp Condition="'$(Temp)' == ''">..\..\..\temp</Temp>
   </PropertyGroup>
```
Comment (jtkech) [inline src/OrchardCore/OrchardCore.AsModule/OrchardCore.AsModule.props]: AsModule is also provided as a package which embeds this .props file. So, any user project and at any place can reference this package and then use this .props file.

So, here the problem i see is to use a path relative to the project being built, we would need an absolute path. Maybe with something like `$(SolutionDir)`, but through the command line this one is `Undefined` ...

Comment (kerryjiang) [inline src/OrchardCore/OrchardCore.AsModule/OrchardCore.AsModule.props]: Yes, probably you just need add something in the document to say additional argument is required when you build the application with command line in osx/linux:

`
dotnet msbuild /t:build /p:Temp:xxxx
`

Comment (jtkech) [inline src/OrchardCore/OrchardCore.AsModule/OrchardCore.AsModule.props]: Yes, good idea and just checked that this property is also passed to the referenced projects when they are built.

So, modules use the Temp folder to store their assets, then notice that the Temp folder is also used by the main app through `AsBundle.targets`, this to retrieve all stored assets. So, defining the Temp variable in AsModule / AsTheme would not be sufficient.

But, as you suggested, using a global property through the command line to define the Temp property should work.

Also, would be good to get rid of the Temp folder as suggested in #707, for this i will continue the discusion on your issue #708 where i will have some questions.

**Pull request closed without merging**